lib/ansible/module_utils/urls.py

```diff
@@ -1145,6 +1146,10 @@ def open(self, method, url, data=None, headers=None, use_proxy=None,
 
         method = method.upper()
 
+        logger = logging.getLogger(__name__)
+        logger.setLevel(logging.DEBUG)
+        logger.addHandler(logging.StreamHandler(sys.stdout))
```

Comment:
We avoid creating loggers for each system, the module class already has it's own logger.

Reply:
@bcoca Can you point me to what you mean? I was looking for a logger, couldn't find one.

Reply:
The `module` object has methods like `info()`, `debug()`, `warn()` that you can use. These details are sent to the syslogger defined by the ansible config.

Reply:
@jborean93 Thank you, I see what you mean. And is it "acceptable" to pass `module` to `def open(...)`?

Reply:
So, it seems like this is not easily doable. Also, probably a hefty BC break if I change the API of `open()`. Can anyone give me a hint how to use module there? The class `Request` is currently unaware of it. Can I inject it in `__init__` or what would you prefer?

Reply:
@jborean93 @bcoca Any feedback? Love to finish this and close this out.

Reply:
The code in urls.py has no access to the module object as you mention, nor should it have access as it's not required to run in a module context.  This puts us in a difficult place, as we cannot and should not have access to it, and just using `logging` is not correct.

We would likely need to abstract the syslog functionality out of the `AnsibleModule` class, similarly to what we did with warnings, and then allow `AnsibleModule` to consume or augment that functionality.  This is likely going to be a large undertaking.

As such, I don't see us adding this functionality right now.

Reply:
Do you think we can turn off netrc then?

This was a huge time sync, trying to figure out what happens so deep in internals.

Reply:
We may be able to add an option that can disable netrc, with the default still to use netrc.  I'm not sure that helps you though.  Although I'm not sure how disabling netrc at this point helps you.